tests/unit/acl-v2.tcl

```diff
@@ -131,11 +131,99 @@ start_server {tags {"acl external:skip"}} {
         catch {$r2 copy read write} err
         assert_match "*NOPERM*keys*" $err
         catch {$r2 copy write read} err
-        assert_match "*NOPERM*keys*" $err   
+        assert_match "*NOPERM*keys*" $err
+    }
+
+    test {Test SET with separate read permission} {
+        r del readstr
+        r ACL SETUSER set-key-permission-R on nopass %R~read* +@all
+        $r2 auth set-key-permission-R password
+        assert_equal PONG [$r2 PING]
+        assert_equal {} [$r2 get readstr]
+
+        # We don't have the permission to WRITE key.
+        assert_error {*NOPERM*keys*} {$r2 set readstr bar}
+        assert_error {*NOPERM*keys*} {$r2 set readstr bar get}
+        assert_error {*NOPERM*keys*} {$r2 set readstr bar ex 100}
+        assert_error {*NOPERM*keys*} {$r2 set readstr bar keepttl nx}
+    }
+
+    test {Test SET with separate write permission} {
+        r del writestr
+        r ACL SETUSER set-key-permission-W on nopass %W~write* +@all
+        $r2 auth set-key-permission-W password
+        assert_equal PONG [$r2 PING]
+        assert_equal {OK} [$r2 set writestr bar]
+        assert_equal {OK} [$r2 set writestr get]
+
+        # We don't have the permission to READ key.
+        assert_error {*NOPERM*keys*} {$r2 set get writestr}
+        assert_error {*NOPERM*keys*} {$r2 set writestr bar get}
+        assert_error {*NOPERM*keys*} {$r2 set writestr bar get ex 100}
+        assert_error {*NOPERM*keys*} {$r2 set writestr bar get keepttl nx}
+
+        # this probably should be `ERR value is not an integer or out of range`
+        assert_error {*NOPERM*keys*} {$r2 set writestr bar ex get}
+    }
+
+    test {Test SET with read and write permissions} {
```

Comment:
This testing seems excessive for the SET and BITFIELD cases. It's doing a lot of validation that the ACLs are working as expected and the commands execute correctly.

Reply:
you mean to use DRYRUN instead of AUTH (i suppose it can be shorter, but i saw it was copied from the test above, so decided to let it go)
or other things?
we do need to test all 3 cases, since we don't have other ways to check that the flags are retrieved correctly.
and also test that if we use the `GET` keyword in the wrong place it doesn't lead to the wrong extraction of flags.